<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 The hardcode - the beta version of the dashboard | A dashboard for data analysis</title>
  <meta name="description" content="RNA seq." />
  <meta name="generator" content="bookdown 0.21.6 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 The hardcode - the beta version of the dashboard | A dashboard for data analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA seq." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 The hardcode - the beta version of the dashboard | A dashboard for data analysis" />
  
  <meta name="twitter:description" content="RNA seq." />
  

<meta name="author" content="Sanne ne Jelmer" />


<meta name="date" content="2021-01-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="summary.html"/>
<link rel="next" href="inleiding-shiny-dashboard.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="summary.html"><a href="summary.html"><i class="fa fa-check"></i><b>3</b> Summary</a></li>
<li class="chapter" data-level="4" data-path="the-hardcode-the-beta-version-of-the-dashboard.html"><a href="the-hardcode-the-beta-version-of-the-dashboard.html"><i class="fa fa-check"></i><b>4</b> The hardcode - the beta version of the dashboard</a><ul>
<li class="chapter" data-level="4.1" data-path="the-hardcode-the-beta-version-of-the-dashboard.html"><a href="the-hardcode-the-beta-version-of-the-dashboard.html#get-the-first-data-of-the-experiment"><i class="fa fa-check"></i><b>4.1</b> Get the first data of the experiment</a></li>
<li class="chapter" data-level="4.2" data-path="the-hardcode-the-beta-version-of-the-dashboard.html"><a href="the-hardcode-the-beta-version-of-the-dashboard.html#untar-the-data."><i class="fa fa-check"></i><b>4.2</b> Untar the data.</a></li>
<li class="chapter" data-level="4.3" data-path="the-hardcode-the-beta-version-of-the-dashboard.html"><a href="the-hardcode-the-beta-version-of-the-dashboard.html#unzipping-of-the-downloaded-data"><i class="fa fa-check"></i><b>4.3</b> Unzipping of the downloaded data</a></li>
<li class="chapter" data-level="4.4" data-path="the-hardcode-the-beta-version-of-the-dashboard.html"><a href="the-hardcode-the-beta-version-of-the-dashboard.html#make-data-ready-for-summarizedexperiment-and-deseq2"><i class="fa fa-check"></i><b>4.4</b> Make data ready for SummarizedExperiment and DESeq2</a><ul>
<li class="chapter" data-level="4.4.1" data-path="the-hardcode-the-beta-version-of-the-dashboard.html"><a href="the-hardcode-the-beta-version-of-the-dashboard.html#genarating-a-summarized-experiment-for-csv"><i class="fa fa-check"></i><b>4.4.1</b> Genarating a summarized experiment for (CSV)</a></li>
<li class="chapter" data-level="4.4.2" data-path="the-hardcode-the-beta-version-of-the-dashboard.html"><a href="the-hardcode-the-beta-version-of-the-dashboard.html#last-steps-before-summarizedexperiment"><i class="fa fa-check"></i><b>4.4.2</b> Last steps before SummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="the-hardcode-the-beta-version-of-the-dashboard.html"><a href="the-hardcode-the-beta-version-of-the-dashboard.html#summarizedexperiment"><i class="fa fa-check"></i><b>4.5</b> SummarizedExperiment</a><ul>
<li class="chapter" data-level="4.5.1" data-path="the-hardcode-the-beta-version-of-the-dashboard.html"><a href="the-hardcode-the-beta-version-of-the-dashboard.html#isee"><i class="fa fa-check"></i><b>4.5.1</b> iSEE</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="the-hardcode-the-beta-version-of-the-dashboard.html"><a href="the-hardcode-the-beta-version-of-the-dashboard.html#deseq2-analysis"><i class="fa fa-check"></i><b>4.6</b> DESeq2 analysis</a><ul>
<li class="chapter" data-level="4.6.1" data-path="the-hardcode-the-beta-version-of-the-dashboard.html"><a href="the-hardcode-the-beta-version-of-the-dashboard.html#the-visualization-of-the-dataset"><i class="fa fa-check"></i><b>4.6.1</b> The visualization of the dataset</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html"><i class="fa fa-check"></i><b>5</b> inleiding shiny dashboard</a><ul>
<li class="chapter" data-level="5.1" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#download-function"><i class="fa fa-check"></i><b>5.1</b> Download function</a><ul>
<li class="chapter" data-level="5.1.1" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#the-working-code-behind-the-application."><i class="fa fa-check"></i><b>5.1.1</b> The working code behind the application.</a></li>
<li class="chapter" data-level="5.1.2" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#shiny"><i class="fa fa-check"></i><b>5.1.2</b> Shiny</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#downloading-supfiles-and-untarunzip-functions."><i class="fa fa-check"></i><b>5.2</b> Downloading supfiles and untar/unzip functions.</a><ul>
<li class="chapter" data-level="5.2.1" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#the-working-code-behind-the-application.-1"><i class="fa fa-check"></i><b>5.2.1</b> The working code behind the application.</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#supplementary-files-download-and-make-assaydata-tab"><i class="fa fa-check"></i><b>5.3</b> Supplementary files download and make assaydata tab</a><ul>
<li class="chapter" data-level="5.3.1" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#shiny-1"><i class="fa fa-check"></i><b>5.3.1</b> Shiny</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#groupselection-tab"><i class="fa fa-check"></i><b>5.4</b> Groupselection tab</a></li>
<li class="chapter" data-level="5.5" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#build-summarized-experiment-function"><i class="fa fa-check"></i><b>5.5</b> Build summarized experiment function</a><ul>
<li class="chapter" data-level="5.5.1" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#the-working-code-behind-the-application.-2"><i class="fa fa-check"></i><b>5.5.1</b> The working code behind the application.</a></li>
<li class="chapter" data-level="5.5.2" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#shiny-2"><i class="fa fa-check"></i><b>5.5.2</b> Shiny</a></li>
<li class="chapter" data-level="5.5.3" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#make-deseq2-tab"><i class="fa fa-check"></i><b>5.5.3</b> Make DESeq2 tab</a></li>
<li class="chapter" data-level="5.5.4" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#figures"><i class="fa fa-check"></i><b>5.5.4</b> Figures</a></li>
<li class="chapter" data-level="5.5.5" data-path="inleiding-shiny-dashboard.html"><a href="inleiding-shiny-dashboard.html#download-data-tab"><i class="fa fa-check"></i><b>5.5.5</b> Download data tab</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion-and-discusion-.html"><a href="conclusion-and-discusion-.html"><i class="fa fa-check"></i><b>6</b> conclusion and discusion.</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A dashboard for data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="the-hardcode---the-beta-version-of-the-dashboard" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> The hardcode - the beta version of the dashboard</h1>
<p>In this part we'll explain each part of our version of the dashboard. This chapter contains the following sub-chapters</p>
<p>(hier dingen zoeken om het anders weer te geven, ook linkjes maken zodat je meteen naar dit hoofdstuk kan etc etc)</p>
<p>Get the first data of the experiment : in this chapter the following things will be discussed and explained...</p>
<p>Untar the data.</p>
<p>Unzipping of the downloaded data</p>
<p>Make data ready for SummarizedExperiment and DESeq2</p>
<p>Genarating a summarized experiment for (CSV)</p>
<p>SummarizedExperiment (S) iSEE</p>
<p>DESeq2 analysis</p>
<p>The visualization of the dataset Heatmap PCA Histogram Vulcan-plot</p>

<div id="get-the-first-data-of-the-experiment" class="section level2">
<h2><span class="header-section-number">4.1</span> Get the first data of the experiment</h2>
<p>With the functions below the data from the selected experiment will be downloaded. The first steps are to collect the data of the experiment and the supplementary files (supp_files from now on). Inside the supp_files are the raw data of the experimnet, a supp_file can be any kind of file. For our project we've focused on csv files only.</p>
<p>Before we can download the files we need to make a directory, this is for... . After that we need to make sure R knows what we want to work with. We can do this with PARAMS, you can find them at the top of the Rmarkdown documents. The params we use are geo_dataset(put the number of your experiemnt from NCBI_GEO (LINKJE)), params data1 and data2 will be explained later. When you want to analyze several experiments at the same time (not recommended) R needs to split the experiments, so it won't read the params as 1 experiment. We do this with the first code chunk, in this chunk, folders are also made for the (different) experiments</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#make a directory</span>
<span class="co">#create_dir &lt;- function(path) {</span>
 <span class="co"># </span>
<span class="co">#}</span>

<span class="co">#When there are more params the (function) &quot;datasets&quot; will split them</span>
datasets &lt;-<span class="st"> </span><span class="kw">strsplit</span>(params<span class="op">$</span>geo_dataset, <span class="dt">split =</span> <span class="st">&quot; &quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>unlist
<span class="co">#With full_paths each parameter will be </span>
full_paths &lt;-<span class="st"> </span><span class="kw">file.path</span>(here<span class="op">::</span><span class="kw">here</span>(), datasets)

<span class="co">#Make maps for the different experiments inside geo_dataset</span>
purrr<span class="op">::</span><span class="kw">walk</span>(
  full_paths,
  dir.create)</code></pre></div>
<p>Now R knows with what kind of data we want to work with, we can download the first set of data. We do this with the function “map2” the .x tells what data to look for and .y where to put it. The function getGEO does all the magic to get the data. You might notice at .y we used full_paths, we made this directory in the last code chunk, it gives the location of the folder for each experiment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#download the data </span>
data &lt;-<span class="st"> </span><span class="kw">map2</span>(
  <span class="dt">.x  =</span> datasets,
  <span class="dt">.y =</span> full_paths,
  getGEO,
  <span class="dt">GSEMatrix =</span> <span class="ot">TRUE</span>,
  <span class="dt">filename =</span> <span class="ot">NULL</span>
)</code></pre></div>
<pre><code>## Found 1 file(s)</code></pre>
<pre><code>## GSE150646_series_matrix.txt.gz</code></pre>
<pre><code>## 
## ── Column specification ────────────────────────────────────────────────────────
## cols(
##   .default = col_character()
## )
## ℹ Use `spec()` for the full column specifications.</code></pre>
<pre><code>## File stored at:</code></pre>
<pre><code>## /home/jelmer_oedzes/New-eindverslag/New_new_research_project/GSE150646/GPL18694.soft</code></pre>
<p>With the first data, about the experiment, downloaded, it’s time to get the supp_files. We use the same map2 function from before but now with the function “getGEOSuppFiles”. This will get the supp_files from the selected experiment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Get supp files</span>
data_supp &lt;-<span class="st"> </span><span class="kw">map2</span>(
  <span class="dt">.x  =</span> datasets,
  <span class="dt">.y =</span> full_paths,
  getGEOSuppFiles,
  <span class="dt">makeDirectory =</span> <span class="ot">FALSE</span>, 
  <span class="co">#baseDir = getwd(),</span>
  <span class="dt">fetch_files =</span> <span class="ot">TRUE</span>, 
  <span class="dt">filter_regex =</span> <span class="ot">NULL</span>
)</code></pre></div>
<p>Now we’ve downloaded the experimental data (supp_files), we need to unpack them. Supp_files are often zipped or tarred, our next job is to unzip and or untar these files. First we’ll explain the untarring of files then after that the unzipping of the data.</p>

</div>
<div id="untar-the-data." class="section level2">
<h2><span class="header-section-number">4.2</span> Untar the data.</h2>
<p>To untar the data we first need to tell R where to find the tarred files. We do this with the first line of code, this contains the location of the files. We only select the tarred files with the pattern function. We tell the map function with .x where to look for the tarred data and with .f to untar these files. The last thing we tell map to do is making a new folder. We do this with exdir, this is needed because in a tarred file there are several smaller files, they need a place to go.</p>
<p>Because the location is now different, we made an unzip function inside the untar code chunk.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#To untar more files (from different geo_datasets doesn&#39;t work yet)</span>
files_tar &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="dt">path =</span> here<span class="op">::</span><span class="kw">here</span>(datasets), <span class="dt">full.names =</span> <span class="ot">TRUE</span>, <span class="dt">pattern =</span> <span class="st">&quot;.tar&quot;</span>)

untarr &lt;-<span class="st"> </span><span class="kw">map</span>(
  <span class="dt">.x =</span> files_tar,
  <span class="dt">.f =</span> untar,
  <span class="dt">exdir =</span> <span class="kw">file.path</span>(root, params<span class="op">$</span>geo_dataset , <span class="st">&quot;untarred&quot;</span>)
)
## Untarred files will be unzipped    ##voor nu gebruiken &quot;Probleem&quot;  normaal gebruiken &gt; datasets (zonder &quot;&quot;) (weet niet meer waarom dit er staat en of het is opgelost)
list_for_unzipping &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="kw">file.path</span>(root, datasets, <span class="st">&quot;untarred&quot;</span>), <span class="dt">full.names =</span> <span class="ot">TRUE</span>)
<span class="kw">lapply</span>(list_for_unzipping, gunzip)</code></pre></div>
<pre><code>## list()</code></pre>

</div>
<div id="unzipping-of-the-downloaded-data" class="section level2">
<h2><span class="header-section-number">4.3</span> Unzipping of the downloaded data</h2>
<p>We start the same way as “untar the data”, we tell R where to find the zipped files and select them with the pattern “.gz”. The new map function will then unzip the files with function GEOquery::gunzip</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">files_zip &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="dt">path =</span> here<span class="op">::</span><span class="kw">here</span>(datasets), <span class="dt">full.names =</span> <span class="ot">TRUE</span>, <span class="dt">pattern =</span> <span class="st">&quot;.gz&quot;</span>)

unzipp &lt;-<span class="st"> </span><span class="kw">map</span>(
  <span class="dt">.x =</span> files_zip,
  <span class="dt">.f =</span> GEOquery<span class="op">::</span>gunzip
)</code></pre></div>
<p>In these first few steps we have: Downloaded supp_files and data about the experiment and unpacked these files. The next steps will be making sure the data is ready for data-analysis.</p>

</div>
<div id="make-data-ready-for-summarizedexperiment-and-deseq2" class="section level2">
<h2><span class="header-section-number">4.4</span> Make data ready for SummarizedExperiment and DESeq2</h2>
<div id="genarating-a-summarized-experiment-for-csv" class="section level3">
<h3><span class="header-section-number">4.4.1</span> Genarating a summarized experiment for (CSV)</h3>
<p>The first steps in making a summarized experiment are getting the phenodata and metadata. Before we do that we first check if we still have the correct data selected. The glimpse(datasets) should give the same output as your selected experiment (params geo_dataset) in this case that is GSE150646.</p>
<p>After this quick check we want to know how to get to our data and make this easier. We see that that the data is saved in a folder inside a folder. To get to the data we need to do the same function twice, that’s why we called the first one 'pre'. Now we can easily ge our data with gse_csv.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#We look into the datasets to check if we have selected the experiments we want to work with. </span>
<span class="kw">glimpse</span>(datasets)</code></pre></div>
<pre><code>##  chr &quot;GSE150646&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">glimpse</span>(data)</code></pre></div>
<pre><code>## List of 1
##  $ :List of 1
##   ..$ GSE150646_series_matrix.txt.gz:Formal class &#39;ExpressionSet&#39; [package &quot;Biobase&quot;] with 7 slots</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#To get to the data we need to do the same function twice, thats why we called the first one &#39;pre&#39;</span>
pre_gse_csv &lt;-<span class="st"> </span>data[[<span class="dv">1</span>]]
gse_csv &lt;-<span class="st"> </span>pre_gse_csv[[<span class="dv">1</span>]]
<span class="kw">glimpse</span>(gse_csv)</code></pre></div>
<pre><code>## Formal class &#39;ExpressionSet&#39; [package &quot;Biobase&quot;] with 7 slots
##   ..@ experimentData   :Formal class &#39;MIAME&#39; [package &quot;Biobase&quot;] with 13 slots
##   ..@ assayData        :&lt;environment: 0x55bb2bbdc9a8&gt; 
##   ..@ phenoData        :Formal class &#39;AnnotatedDataFrame&#39; [package &quot;Biobase&quot;] with 4 slots
##   ..@ featureData      :Formal class &#39;AnnotatedDataFrame&#39; [package &quot;Biobase&quot;] with 4 slots
##   ..@ annotation       : chr &quot;GPL18694&quot;
##   ..@ protocolData     :Formal class &#39;AnnotatedDataFrame&#39; [package &quot;Biobase&quot;] with 4 slots
##   ..@ .__classVersion__:Formal class &#39;Versions&#39; [package &quot;Biobase&quot;] with 1 slot</code></pre>
<p>Now we know that we’ve the correct data and can easily access them we’ll make the phenodata and metadata. We only need 2 functions to do this, pData and experimentData.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#obtaining phenotypic data. contains all the info regarding each sample. #werkt niet</span>
phenodata_csv &lt;-<span class="st"> </span><span class="kw">pData</span>(gse_csv) <span class="op">%&gt;%</span><span class="st"> </span>as_tibble
<span class="kw">class</span>(phenodata_csv)</code></pre></div>
<pre><code>## [1] &quot;tbl_df&quot;     &quot;tbl&quot;        &quot;data.frame&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#metadata maken en controleren of dit klopt</span>
metadata_csv &lt;-<span class="st"> </span><span class="kw">experimentData</span>(gse_csv)
<span class="co">#class(metadata_csv)</span></code></pre></div>
<p>Now we need to “clean up” the phenodata a little bit before we can continue making a summarizedExperiment. When we look at the phenodata (with head) we see some weird names like “characteristics_ch1.1” and “characteristics_ch1.2”, these are the variables of the experiments. When you read “characteristics_ch1.1” you’ve no clue what the experiment is about, so you can change the name (for example) to gender/dev_stage/genotype etc. You can change the names with the params data1 and data2 at the start of this document. The place of “characteristics_ch1.1” can change from experiment to experiment, we haven’t find a way to automate this yet.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#To look at what columm (number) the characteristics are, this can change per research, we haven&#39;t automated this yet.</span>
<span class="kw">names</span>(phenodata_csv)</code></pre></div>
<pre><code>##  [1] &quot;title&quot;                   &quot;geo_accession&quot;          
##  [3] &quot;status&quot;                  &quot;submission_date&quot;        
##  [5] &quot;last_update_date&quot;        &quot;type&quot;                   
##  [7] &quot;channel_count&quot;           &quot;source_name_ch1&quot;        
##  [9] &quot;organism_ch1&quot;            &quot;characteristics_ch1&quot;    
## [11] &quot;characteristics_ch1.1&quot;   &quot;characteristics_ch1.2&quot;  
## [13] &quot;treatment_protocol_ch1&quot;  &quot;molecule_ch1&quot;           
## [15] &quot;extract_protocol_ch1&quot;    &quot;extract_protocol_ch1.1&quot; 
## [17] &quot;extract_protocol_ch1.2&quot;  &quot;taxid_ch1&quot;              
## [19] &quot;data_processing&quot;         &quot;data_processing.1&quot;      
## [21] &quot;data_processing.2&quot;       &quot;data_processing.3&quot;      
## [23] &quot;platform_id&quot;             &quot;contact_name&quot;           
## [25] &quot;contact_email&quot;           &quot;contact_department&quot;     
## [27] &quot;contact_institute&quot;       &quot;contact_address&quot;        
## [29] &quot;contact_city&quot;            &quot;contact_zip/postal_code&quot;
## [31] &quot;contact_country&quot;         &quot;data_row_count&quot;         
## [33] &quot;instrument_model&quot;        &quot;library_selection&quot;      
## [35] &quot;library_source&quot;          &quot;library_strategy&quot;       
## [37] &quot;relation&quot;                &quot;relation.1&quot;             
## [39] &quot;supplementary_file_1&quot;    &quot;age:ch1&quot;                
## [41] &quot;genotype:ch1&quot;            &quot;strain:ch1&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Het benoemen van experimentele waarde</span>
<span class="kw">colnames</span>(phenodata_csv)[<span class="dv">11</span>] &lt;-<span class="st"> </span>params<span class="op">$</span>data1
<span class="kw">colnames</span>(phenodata_csv)[<span class="dv">12</span>] &lt;-<span class="st"> </span>params<span class="op">$</span>data2


<span class="kw">names</span>(phenodata_csv)</code></pre></div>
<pre><code>##  [1] &quot;title&quot;                   &quot;geo_accession&quot;          
##  [3] &quot;status&quot;                  &quot;submission_date&quot;        
##  [5] &quot;last_update_date&quot;        &quot;type&quot;                   
##  [7] &quot;channel_count&quot;           &quot;source_name_ch1&quot;        
##  [9] &quot;organism_ch1&quot;            &quot;characteristics_ch1&quot;    
## [11] &quot;dev_stage&quot;               &quot;genotype&quot;               
## [13] &quot;treatment_protocol_ch1&quot;  &quot;molecule_ch1&quot;           
## [15] &quot;extract_protocol_ch1&quot;    &quot;extract_protocol_ch1.1&quot; 
## [17] &quot;extract_protocol_ch1.2&quot;  &quot;taxid_ch1&quot;              
## [19] &quot;data_processing&quot;         &quot;data_processing.1&quot;      
## [21] &quot;data_processing.2&quot;       &quot;data_processing.3&quot;      
## [23] &quot;platform_id&quot;             &quot;contact_name&quot;           
## [25] &quot;contact_email&quot;           &quot;contact_department&quot;     
## [27] &quot;contact_institute&quot;       &quot;contact_address&quot;        
## [29] &quot;contact_city&quot;            &quot;contact_zip/postal_code&quot;
## [31] &quot;contact_country&quot;         &quot;data_row_count&quot;         
## [33] &quot;instrument_model&quot;        &quot;library_selection&quot;      
## [35] &quot;library_source&quot;          &quot;library_strategy&quot;       
## [37] &quot;relation&quot;                &quot;relation.1&quot;             
## [39] &quot;supplementary_file_1&quot;    &quot;age:ch1&quot;                
## [41] &quot;genotype:ch1&quot;            &quot;strain:ch1&quot;</code></pre>
<p>Further change of the phnodata, needed to make Summarized Experiment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#</span>
phenodata_csv &lt;-<span class="st"> </span>phenodata_csv[<span class="dv">2</span><span class="op">:</span><span class="dv">41</span>]
samplenames_csv&lt;-<span class="st"> </span>phenodata_csv[<span class="dv">1</span>]</code></pre></div>

</div>
<div id="last-steps-before-summarizedexperiment" class="section level3">
<h3><span class="header-section-number">4.4.2</span> Last steps before SummarizedExperiment</h3>
<p>Now we can load the CSV_file, you’ll see the first few counts inside the file. Check if the data file is a “data.frame.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Load the raw data from the csv data</span>
load_csv_data &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="dt">path =</span> here<span class="op">::</span><span class="kw">here</span>(params<span class="op">$</span>geo_dataset), <span class="dt">full.names =</span> <span class="ot">TRUE</span>, <span class="dt">pattern =</span> <span class="st">&quot;.csv&quot;</span>)
csv_data &lt;-<span class="st"> </span><span class="kw">read.csv</span>(load_csv_data)
<span class="co">#the first few counts of the raw data</span>
<span class="kw">head</span>(csv_data)</code></pre></div>
<pre><code>##       Ensembl_geneID I16R020c01 I16R020c03 I16R020c04 I16R020c05 I16R020c06
## 1 ENSRNOG00000040300          0          1          0          1          1
## 2 ENSRNOG00000061316          7          5          1          6          1
## 3 ENSRNOG00000050129          0          0          1          0          0
## 4 ENSRNOG00000029897          0          0          0          2          0
## 5 ENSRNOG00000042852          0          2          0          0          0
## 6 ENSRNOG00000014303       4001       3825       3810       4494       3934
##   I16R020d01 I16R020d02 I16R020d03 I16R020d04 I16R020d06 I16R043g01 I16R043g04
## 1          2          0          0          0          2          0          1
## 2          1          2          1          2          2         10          6
## 3          0          0          0          0          0          0          0
## 4          1          1          0          2          1          1          4
## 5          0          0          0          0          0          0          1
## 6       4014       4167       4254       4811       4462       4406       4883
##   I16R043g05 I16R043g06 I16R043g07 I16R043h02 I16R043h03 I16R043h05 I16R043h06
## 1          0          2          1          0          0          2          1
## 2          2          2          6          3          4          8          7
## 3          0          0          0          0          0          2          0
## 4          1          2          4          5          0          5          6
## 5          0          0          0          0          0          0          1
## 6       5092       4935       5546       5870       5646       5316       4742
##   I16R043h07
## 1          0
## 2          2
## 3          0
## 4          4
## 5          0
## 6       5534</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Check if the csv_data is a data.frame</span>
<span class="kw">class</span>(csv_data)</code></pre></div>
<pre><code>## [1] &quot;data.frame&quot;</code></pre>
<p>In raw_counts_csv_data we load in the raw counts of the csv file, minus the first column because these are names of the samples. Also note that it can be more or less columns in different documents. In the next step we change the name of the columns to make them more understandable. The last step is to change the data.frame into a data.matrix for SummarizedExperiment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## In het volgende stukje willen we dat de namen goed komen te staan, je ziet eerst dat het rommelig is en daarna hebben ze GSM nummers
<span class="co">#Laat alleen de raw counts over</span>
raw_counts_csv_data &lt;-<span class="st"> </span>csv_data[<span class="dv">2</span><span class="op">:</span><span class="dv">21</span>]
<span class="co">#head(raw_counts_csv_data)</span>

##make rowdata_csv
rowdata_csv_data &lt;-<span class="st"> </span>raw_counts_csv_data<span class="op">$</span><span class="st">`</span><span class="dt">Gene symbol</span><span class="st">`</span>

<span class="co">#Kan ik de raw_counts_csv_data gebruiken om colnames te maken test</span>
<span class="kw">colnames</span>(raw_counts_csv_data) &lt;-samplenames_csv[[<span class="dv">1</span>]]
<span class="kw">head</span>(raw_counts_csv_data)</code></pre></div>
<pre><code>##   GSM4555720 GSM4555721 GSM4555722 GSM4555723 GSM4555724 GSM4555725 GSM4555726
## 1          0          1          0          1          1          2          0
## 2          7          5          1          6          1          1          2
## 3          0          0          1          0          0          0          0
## 4          0          0          0          2          0          1          1
## 5          0          2          0          0          0          0          0
## 6       4001       3825       3810       4494       3934       4014       4167
##   GSM4555727 GSM4555728 GSM4555729 GSM4555730 GSM4555731 GSM4555732 GSM4555733
## 1          0          0          2          0          1          0          2
## 2          1          2          2         10          6          2          2
## 3          0          0          0          0          0          0          0
## 4          0          2          1          1          4          1          2
## 5          0          0          0          0          1          0          0
## 6       4254       4811       4462       4406       4883       5092       4935
##   GSM4555734 GSM4555735 GSM4555736 GSM4555737 GSM4555738 GSM4555739
## 1          1          0          0          2          1          0
## 2          6          3          4          8          7          2
## 3          0          0          0          2          0          0
## 4          4          5          0          5          6          4
## 5          0          0          0          0          1          0
## 6       5546       5870       5646       5316       4742       5534</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#maak er een matrix van, weet niet waarom</span>

raw_counts_csv_data_matrix &lt;-<span class="st"> </span><span class="kw">data.matrix</span>(raw_counts_csv_data)
<span class="kw">rownames</span>(raw_counts_csv_data_matrix) &lt;-<span class="st"> </span>rowdata_csv_data
<span class="co">#head(raw_counts_csv_data_matrix)</span></code></pre></div>

</div>
</div>
<div id="summarizedexperiment" class="section level2">
<h2><span class="header-section-number">4.5</span> SummarizedExperiment</h2>
<p>SummarizedExperiment is used for organising data. This package saves rectangular matrices from experimental results. The rows represent the characteristics of interest, typically genes and transcripts. The columns stand for the samples. ColData consits of samples and metadata. Metadata is data about data, so ColData contains extra information about the samples.</p>
<p>Here we will put a character factor in a word. The word now consits of the organised data of the raw sample data coming from the dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">se_csv_data &lt;-<span class="st"> </span><span class="kw">SummarizedExperiment</span>(<span class="dt">assays =</span> raw_counts_csv_data_matrix,
                     <span class="dt">rowData =</span> rowdata_csv_data,
                     <span class="dt">colData =</span> phenodata_csv)</code></pre></div>
<p>Here we will ad the metadata manually because the SummarizedExperiment won't accept it otherwise. The $ function extracts elements from a named list, by heading.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">metadata</span>(se_csv_data)<span class="op">$</span>metadata &lt;-<span class="st"> </span>metadata </code></pre></div>
<p>With the code below we will be able to inspect the SummarizedExperiment. The code &quot;head&quot; will make it possible to view the first part of the data frame.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">se_csv_data</code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 22299 20 
## metadata(1): metadata
## assays(1): &#39;&#39;
## rownames: NULL
## rowData names(0):
## colnames(20): GSM4555720 GSM4555721 ... GSM4555738 GSM4555739
## colData names(40): geo_accession status ... age:ch1 genotype:ch1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw">assay</span>(se_csv_data))</code></pre></div>
<pre><code>##      GSM4555720 GSM4555721 GSM4555722 GSM4555723 GSM4555724 GSM4555725
## [1,]          0          1          0          1          1          2
## [2,]          7          5          1          6          1          1
## [3,]          0          0          1          0          0          0
## [4,]          0          0          0          2          0          1
## [5,]          0          2          0          0          0          0
## [6,]       4001       3825       3810       4494       3934       4014
##      GSM4555726 GSM4555727 GSM4555728 GSM4555729 GSM4555730 GSM4555731
## [1,]          0          0          0          2          0          1
## [2,]          2          1          2          2         10          6
## [3,]          0          0          0          0          0          0
## [4,]          1          0          2          1          1          4
## [5,]          0          0          0          0          0          1
## [6,]       4167       4254       4811       4462       4406       4883
##      GSM4555732 GSM4555733 GSM4555734 GSM4555735 GSM4555736 GSM4555737
## [1,]          0          2          1          0          0          2
## [2,]          2          2          6          3          4          8
## [3,]          0          0          0          0          0          2
## [4,]          1          2          4          5          0          5
## [5,]          0          0          0          0          0          0
## [6,]       5092       4935       5546       5870       5646       5316
##      GSM4555738 GSM4555739
## [1,]          1          0
## [2,]          7          2
## [3,]          0          0
## [4,]          6          4
## [5,]          1          0
## [6,]       4742       5534</code></pre>
<div id="isee" class="section level3">
<h3><span class="header-section-number">4.5.1</span> iSEE</h3>
<p>This function can be used to explore the data stored in the SummarizedExperiment, like the row- and column-level metadata.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#iSEE test</span>
<span class="co">#se_csv_iSEE &lt;- se_csv_data</span>
<span class="co">#iSEE(se_csv_iSEE)</span></code></pre></div>

</div>
</div>
<div id="deseq2-analysis" class="section level2">
<h2><span class="header-section-number">4.6</span> DESeq2 analysis</h2>
<p>DESeq2 is a {package} that will be used to detect the dfference between genes and to visualize the data. The raw count data, so the raw samples, will be used as input.</p>
<p>The code below still has to be seperated for now it doesn't work proprely with the parameters. We are looking into methods to solve this issue. This function still needs to be manually changed with processing every new dataset. The non-working code is turned off. With this function the DESeq2 dataset will be created.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#condition &lt;- (get(params$data1, params$data2))</span>
<span class="co">#ddsSE_csv_data &lt;- DESeq2::DESeqDataSet(se_csv_data, design = formula(~{{params$data1}}, {{params$data2}}))</span>
ddsSE_csv_data &lt;-<span class="st"> </span>DESeq2<span class="op">::</span><span class="kw">DESeqDataSet</span>(se_csv_data, <span class="dt">design =</span> <span class="kw">formula</span>(<span class="op">~</span>dev_stage, genotype))</code></pre></div>
<pre><code>## renaming the first element in assays to &#39;counts&#39;</code></pre>
<pre><code>##   Note: levels of factors in the design contain characters other than
##   letters, numbers, &#39;_&#39; and &#39;.&#39;. It is recommended (but not required) to use
##   only letters, numbers, and delimiters &#39;_&#39; or &#39;.&#39;, as these are safe characters
##   for column names in R. [This is a message, not an warning or error]</code></pre>
<p>The function rowSums returns the sums of each row in the data set. The counts function will quickly count the unique values of more variables.</p>
<p>In the code chunck below the sums of the rows of the unique values of more variables in the DESeq2 data set are calculated. The calculated rows and the already calculated DESeq2 dataset are put in the word ddsSE_csv_data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">keep &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">counts</span>(ddsSE_csv_data)) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">10</span>
ddsSE_csv_data &lt;-<span class="st"> </span>ddsSE_csv_data[keep,]</code></pre></div>
<p>Below the differntial expression analysis is implemented.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ddsDE_csv_data &lt;-<span class="st"> </span><span class="kw">DESeq</span>(ddsSE_csv_data)</code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>##   Note: levels of factors in the design contain characters other than
##   letters, numbers, &#39;_&#39; and &#39;.&#39;. It is recommended (but not required) to use
##   only letters, numbers, and delimiters &#39;_&#39; or &#39;.&#39;, as these are safe characters
##   for column names in R. [This is a message, not an warning or error]</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>##   Note: levels of factors in the design contain characters other than
##   letters, numbers, &#39;_&#39; and &#39;.&#39;. It is recommended (but not required) to use
##   only letters, numbers, and delimiters &#39;_&#39; or &#39;.&#39;, as these are safe characters
##   for column names in R. [This is a message, not an warning or error]</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>##   Note: levels of factors in the design contain characters other than
##   letters, numbers, &#39;_&#39; and &#39;.&#39;. It is recommended (but not required) to use
##   only letters, numbers, and delimiters &#39;_&#39; or &#39;.&#39;, as these are safe characters
##   for column names in R. [This is a message, not an warning or error]</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##   Note: levels of factors in the design contain characters other than
##   letters, numbers, &#39;_&#39; and &#39;.&#39;. It is recommended (but not required) to use
##   only letters, numbers, and delimiters &#39;_&#39; or &#39;.&#39;, as these are safe characters
##   for column names in R. [This is a message, not an warning or error]</code></pre>
<pre><code>## -- replacing outliers and refitting for 12 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>##   Note: levels of factors in the design contain characters other than
##   letters, numbers, &#39;_&#39; and &#39;.&#39;. It is recommended (but not required) to use
##   only letters, numbers, and delimiters &#39;_&#39; or &#39;.&#39;, as these are safe characters
##   for column names in R. [This is a message, not an warning or error]</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##   Note: levels of factors in the design contain characters other than
##   letters, numbers, &#39;_&#39; and &#39;.&#39;. It is recommended (but not required) to use
##   only letters, numbers, and delimiters &#39;_&#39; or &#39;.&#39;, as these are safe characters
##   for column names in R. [This is a message, not an warning or error]</code></pre>
<p>Here the results of the analysis are transferred into the word res_csv_data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_csv_data &lt;-<span class="st"> </span><span class="kw">results</span>(ddsDE_csv_data)</code></pre></div>
<p>This code will show the top genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_csv_data[<span class="kw">order</span>(res_csv_data<span class="op">$</span>padj), ] <span class="op">%&gt;%</span><span class="st"> </span>head</code></pre></div>
<pre><code>## log2 fold change (MLE): dev stage age..5.months vs age..12.months 
## Wald test p-value: dev stage age..5.months vs age..12.months 
## DataFrame with 6 rows and 6 columns
##           baseMean     log2FoldChange              lfcSE              stat
##          &lt;numeric&gt;          &lt;numeric&gt;          &lt;numeric&gt;         &lt;numeric&gt;
## 1 192.815843867977  -1.24801787212202 0.0817109121555097 -15.2735765542163
## 2 1777.06694763123 -0.590496392470499 0.0410800783419593 -14.3742762015954
## 3 177.261598908197  -1.65284464482474  0.121787508012539 -13.5715449950299
## 4 1524.26391611481 -0.642772186015504 0.0477993506231875 -13.4472995476992
## 5 539.314440529911  -0.66136608389719 0.0528648054944776 -12.5105176820574
## 6 27.0310540415811     -4.20774212721  0.336983986487428 -12.4864750134558
##                 pvalue                 padj
##              &lt;numeric&gt;            &lt;numeric&gt;
## 1 1.14705424786208e-52 2.17687955159265e-48
## 2 7.50505673470035e-47 7.12154833555717e-43
## 3 5.90637914529913e-42 3.73637544731623e-38
## 4 3.19322371432675e-41 1.51502499126233e-37
## 5 6.53966334765167e-36 2.48219462023467e-32
## 6 8.84880714361166e-36 2.79887769952437e-32</code></pre>
<p>The genes will be stored in a top 5 list for later use in the sequence alignment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">top5genes &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_csv_data[<span class="kw">order</span>(res_csv_data<span class="op">$</span>padj), ])[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</code></pre></div>
<p>The log fold change will be calculated and the resultNames will be put into the LFC function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">resultsNames</span>(ddsDE_csv_data)</code></pre></div>
<pre><code>## [1] &quot;Intercept&quot;                                
## [2] &quot;dev_stage_age..5.months_vs_age..12.months&quot;</code></pre>
<p>We also had some problems with automating the code below, this is still a work in progress. A Log 2 fold change is apllied.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resLFC_csv_data &lt;-<span class="st"> </span><span class="kw">lfcShrink</span>(ddsDE_csv_data, 
                             <span class="dt">coef =</span> <span class="st">&quot;dev_stage_age..5.months_vs_age..12.months&quot;</span>     )                       </code></pre></div>
<pre><code>## using &#39;normal&#39; for LFC shrinkage, the Normal prior from Love et al (2014).
## 
## Note that type=&#39;apeglm&#39; and type=&#39;ashr&#39; have shown to have less bias than type=&#39;normal&#39;.
## See ?lfcShrink for more details on shrinkage type, and the DESeq2 vignette.
## Reference: https://doi.org/10.1093/bioinformatics/bty895</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#coef=&quot;genotype_genotype..wildtype_vs_genotype..transgene.overexpressing.human.SNCA&quot;, type=&quot;apeglm&quot;)</span></code></pre></div>

<div id="the-visualization-of-the-dataset" class="section level3">
<h3><span class="header-section-number">4.6.1</span> The visualization of the dataset</h3>
<p>The code below will plot a histogram, this histogram will show the distribution of the p-values calculated by the DESeq2 analyze.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(<span class="kw">as</span>(res_csv_data, <span class="st">&quot;data.frame&quot;</span>), <span class="kw">aes</span>(<span class="dt">x =</span> pvalue)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_histogram</span>()</code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="new_eindverslag_dashboard_sj_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<p>This code will create a volcano plot, a type of scatterplot, that will help by quickly identifying changes in the dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">DESeq2<span class="op">::</span><span class="kw">plotMA</span>(resLFC_csv_data, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">2</span>,<span class="dv">2</span>))</code></pre></div>
<p><img src="new_eindverslag_dashboard_sj_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<p>Here a Principal Component Analysis (PCA) is created, a PCA is used to observe outliers, jumps, clusters and trends.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ddsDE_rlog_csv_data &lt;-<span class="st"> </span>ddsDE_csv_data <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rlogTransformation</span>() 
ddsDE_rlog_csv_data <span class="op">%&gt;%</span>
<span class="st">  </span>DESeq2<span class="op">::</span><span class="kw">plotPCA</span>(<span class="dt">intgroup =</span> <span class="kw">c</span>(params<span class="op">$</span>data1 , params<span class="op">$</span>data2)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> <span class="dv">0</span>, <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="dt">colour =</span> <span class="st">&quot;red&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="dt">colour =</span> <span class="st">&quot;red&quot;</span>)</code></pre></div>
<p><img src="new_eindverslag_dashboard_sj_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<p>The code below creates a heatmap, this shows the extent of a phenomenon of the top 20 of up and down regulated genes as colour in two dimensions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">matrix_results_csv &lt;-<span class="st"> </span><span class="kw">assay</span>(ddsDE_rlog_csv_data)
ind &lt;-<span class="st"> </span>matrix_results_csv <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rowMeans</span>() <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">order</span>(<span class="dt">decreasing =</span> <span class="ot">TRUE</span>) 
top20 &lt;-<span class="st"> </span>matrix_results_csv[ind[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>],]
annot_df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(ddsDE_csv_data)[,<span class="kw">c</span>(params<span class="op">$</span>data1 , params<span class="op">$</span>data2)])
heatmap &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(top20,
                    <span class="dt">scale =</span> <span class="st">&quot;row&quot;</span>, 
                    <span class="dt">annotation_col =</span> annot_df)
heatmap</code></pre></div>
<p><img src="new_eindverslag_dashboard_sj_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="summary.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="inleiding-shiny-dashboard.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["new_eindverslag_dashboard_sj.pdf", "new_eindverslag_dashboard_sj.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
